pgmpy/models/DiscreteBayesianNetwork.py

```diff
@@ -1481,41 +1482,36 @@ def save(self, filename, filetype="bif"):
 
         filetype: str (default: bif)
             The format in which to write the model to file. Can be one of
-            the following: bif, uai, xmlbif, xdsl.
+            the following: bif, uai, xmlbif, xdsl, net.
 
         Examples
         --------
         >>> from pgmpy.utils import get_example_model
         >>> alarm = get_example_model('alarm')
         >>> alarm.save('alarm.bif', filetype='bif')
         """
-        supported_formats = {"bif", "uai", "xmlbif", "xdsl"}
-        if filename.split(".")[-1].lower() in supported_formats:
-            filetype = filename.split(".")[-1].lower()
-
-        if filetype == "bif":
-            from pgmpy.readwrite import BIFWriter
-
-            writer = BIFWriter(self)
-            writer.write_bif(filename=filename)
-
-        elif filetype == "uai":
-            from pgmpy.readwrite import UAIWriter
-
-            writer = UAIWriter(self)
-            writer.write_uai(filename=filename)
+        supported_formats_writer_map = {
+            "bif": ("pgmpy.readwrite", "BIFWriter", "write_bif"),
```

Comment:
Should we also rename the write methods of these classes to just `write` to make it uniform? I think it would simplify the implementation here as well.